inference-engine/tests/functional/plugin/shared/src/behavior/set_blob_of_kind.cpp

```diff
@@ -0,0 +1,139 @@
+    case BlobKind::Compound:
+        return os << "Compound";
+    case BlobKind::BatchOfSimple:
+        return os << "BatchOfSimple";
+    default:
+        THROW_IE_EXCEPTION << "Test does not support the blob kind";
+  }
+}
+
+std::string SetBlobOfKindTest::getTestCaseName(testing::TestParamInfo<SetBlobOfKindParams> obj) {
+    BlobKind blobKind;
+    std::string targetDevice;
+    std::map<std::string, std::string> configuration;
+    std::tie(blobKind, targetDevice, configuration) = obj.param;
+
+    std::ostringstream result;
+    result << "Kind=" << blobKind;
+    result << " Device="<< targetDevice;
+    return result.str();
+}
+
+namespace {
+
+Blob::Ptr makeBlobOfKind(const TensorDesc& td, BlobKind blobKind) {
+    switch (blobKind) {
+    case BlobKind::Simple:
+        return FuncTestUtils::createAndFillBlob(td);
+    case BlobKind::Compound:
+        return make_shared_blob<CompoundBlob>(std::vector<Blob::Ptr>{});
+    case BlobKind::BatchOfSimple: {
+        const auto subBlobsNum = td.getDims()[0];
+        auto subBlobDesc = td;
+        subBlobDesc.getDims()[0] = 1;
+        std::vector<Blob::Ptr> subBlobs;
+        for (size_t i = 0; i < subBlobsNum; i++) {
+            subBlobs.push_back(makeBlobOfKind(subBlobDesc, BlobKind::Simple));
+        }
+        return make_shared_blob<BatchedBlob>(subBlobs);
+    }
+    default:
+        THROW_IE_EXCEPTION << "Test does not support the blob kind";
+    }
+}
+
+bool isBatchedBlobSupported(const std::shared_ptr<Core>& core, const LayerTestsUtils::TargetDevice& targetDevice) {
+    const std::vector<std::string> supported_metrics = core->GetMetric(targetDevice, METRIC_KEY(SUPPORTED_METRICS));
+
+    if (std::find(supported_metrics.begin(), supported_metrics.end(),
+                 METRIC_KEY(OPTIMIZATION_CAPABILITIES)) == supported_metrics.end()) {
+        return false;
+    }
+
+    const std::vector<std::string> optimization_caps =
+        core->GetMetric(targetDevice, METRIC_KEY(OPTIMIZATION_CAPABILITIES));
+
+    return std::find(optimization_caps.begin(), optimization_caps.end(),
+                    METRIC_VALUE(BATCHED_BLOB)) != optimization_caps.end();
+}
+
+bool isBlobKindSupported(const std::shared_ptr<Core>& core, const LayerTestsUtils::TargetDevice& targetDevice, BlobKind blobKind) {
+    switch (blobKind) {
+    case BlobKind::Simple:
+        return true;
+    case BlobKind::Compound:
+        return false;
+    case BlobKind::BatchOfSimple:
+        return isBatchedBlobSupported(core, targetDevice);
+    default:
+        THROW_IE_EXCEPTION << "Test does not support the blob kind";
+    }
+}
+
+} // namespace
+
+Blob::Ptr SetBlobOfKindTest::GenerateInput(const InferenceEngine::InputInfo& info) const {
+    return makeBlobOfKind(info.getTensorDesc(), blobKind);
+}
+
+void SetBlobOfKindTest::Run() {
+    SKIP_IF_CURRENT_TEST_IS_DISABLED()
+
+    LoadNetwork();
+
+    if (isBlobKindSupported(core, targetDevice, blobKind)) {
+        Infer();
+        Validate();
```

Comment:
Do you really want to check plugin inference results? Or Just check behavior (throw exceptions or not)? I think, it should be second option.

Reply:
Yes, if plugin supports a kind of blob, it should handle it properly and produce correct result.

Reply:
In such case these tests are not behavior. Behavior tests provide binary answer: supported or not, exception is thrown or not . If you want to check accuracy, it should be outside of behavior. IMO, I see two testcases - 1. behavior (supported or not) 2. accuracy check

Reply:
done.